### PR TITLE
Adding useGlobalParameters fixture.

### PR DESCRIPTION
**Pull request by sdquiring**
`useGlobalParameters` fixture allows unit testing of components that use `globalParameters`.

Adding unit test of `tryAssembler` function, and refactoring it to handle multi-line inputs.

### CODE REVIEW

**Comment by sdquiring** [comment]
This is looking hopeful to pass CI.

**Comment by sdquiring** [comment]
This is now passing, can someone review this please?

File: Tensile/Common.py [hunk 1]
```diff
@@ -1291,33 +1295,36 @@ def GetArchCaps(isaVersion):
...
-  asmCmd = "%s -x assembler -target amdgcn-amdhsa -mcpu=%s %s -" \
-             % (globalParameters["AssemblerPath"], gfxName(isaVersion), options)
+  args = [globalParameters["AssemblerPath"], '-x', 'assembler',
+          '-target', 'amdgcn-amdhsa',
+          '-mcpu='+gfxName(isaVersion),
+          *options,
+          '-']
 
-  sysCmd = "echo \"%s\" | %s" % (asmString, asmCmd)
+  result = subprocess.run(args, input=asmString.encode(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```
Comment (AlexBrownAMD) [inline Tensile/Common.py:1316]: Is the try/except not needed anymore?

Comment (sdquiring) [reply Tensile/Common.py:1316]: `check_output()` raises an exception if the underlying command returns a non-zero error code.  `run()` by default does not, and just returns an object which has the `returncode` member.

**Pull request merged**